repos/iiif-storage/src/JsonBuilder/CollectionBuilder.php

```diff
@@ -28,31 +31,37 @@ class CollectionBuilder
      * @var string
      */
     private $siteId;
-
-    public function __construct(ApiRouter $router, ManifestBuilder $builder)
-    {
+    /**
+     * @var CollectionRepository
+     */
+    private $collectionRepository;
+    /**
+     * @var ManifestRepository
+     */
+    private $manifestRepository;
+
+    public function __construct(
+        ApiRouter $router,
+        ManifestBuilder $builder,
+        ManifestRepository $manifestRepository,
+        CollectionRepository $collectionRepository
+    ) {
         $this->router = $router;
         $this->manifestBuilder = $builder;
+        $this->collectionRepository = $collectionRepository;
+        $this->manifestRepository = $manifestRepository;
     }
 
     public function buildResource(ItemSetRepresentation $omekaCollection, $originalIds = false): CollectionRepresentation
     {
         $json = $this->build($omekaCollection, $originalIds);
-        // Map children.
-        $manifestIndex = array_reduce(
-            $omekaCollection->value('sc:hasManifests', ['all' => true]),
-            function($acc, ValueRepresentation $next) {
-                /** @var ItemSetRepresentation $manifest */
-                $manifest = $next->valueResource();
-                /** @var ValueRepresentation $identifier */
-                $identifier = $this->router->manifest($manifest->id());
-                $acc[$identifier] = $manifest;
-                return $acc;
-            }, []
-        );
         // Build collection.
-        $collection = ResourceFactory::createCollection($json, function ($url) use ($manifestIndex, $originalIds) {
-            return $this->manifestBuilder->build($manifestIndex[$url], $originalIds);
+        $collection = ResourceFactory::createCollection($json, function (string $url, Manifest $manifest) use ($originalIds) {
+            $metadata = $manifest->getSource();
+            if (!isset($metadata['o:id'])) {
+                return (array)$manifest->getSource();
```

Comment:
This trick here short-cuts loading in ALL of the manifests on the collection. Allows us to lazy-load without issue.